langlib/lang.array/src/main/java/org/ballerinalang/langlib/array/Sort.java

```diff
@@ -42,12 +48,21 @@
 public class Sort {
 
     public static BArray sort(BArray arr, Object direction, Object func) {
-        checkIsArrayOnlyOperation(TypeUtils.getImpliedType(arr.getType()), "sort()");
+        BArray sortedArray;
         BFunctionPointer<Object, Object> function = (BFunctionPointer<Object, Object>) func;
-
+        Set<Type> typeList = new HashSet<>();
+        if (arr.getType().getTag() == TypeTags.TUPLE_TAG) {
```

Comment:
Why only for tuples?

Reply:
Added for arrays as well in c5daaea654d92b306868ce09652d183a764da99f